feat: allow for selector in copy-file-to-pod

dkoshkin opened the pull request:
This is useful when for example copying files to Pods belonging to a Job, which will have a generated name.

Ran the demo (after reverting https://github.com/mesosphere/mindthegap/pull/852) 

```
...
node/self-hosted-registry-demo-control-plane condition met
node/self-hosted-registry-demo-worker condition met
NAME                                      STATUS   ROLES           AGE   VERSION
self-hosted-registry-demo-control-plane   Ready    control-plane   48s   v1.32.2
self-hosted-registry-demo-worker          Ready    <none>          39s   v1.32.2
pod "busybox" deleted
error: timed out waiting for the condition
Yay everything worked as expected - this was supposed to fail!
```

jimmidyson commented on cmd/copy-file-to-pod/main.go:
```diff
@@ -109,6 +117,34 @@ func main() {
+
+	var pod string
+	if podName != "" {
+		pod = podName
```
Let's just return here
```suggestion
		return podName
```

jimmidyson commented on cmd/copy-file-to-pod/main.go:
```diff
@@ -109,6 +117,34 @@ func main() {
+	var pod string
+	if podName != "" {
+		pod = podName
+	} else {
+		pods, err := cs.CoreV1().Pods(namespace).List(
+			context.Background(),
+			metav1.ListOptions{LabelSelector: podSelector},
+		)
+		if err != nil {
+			return "", fmt.Errorf("error listing pods: %w", err)
+		}
+		if len(pods.Items) == 0 {
+			return "", fmt.Errorf("no pods found matching selector %q", podSelector)
+		} else if len(pods.Items) > 1 {
+			return "", fmt.Errorf("multiple pods found matching selector %q, use --pod instead", podSelector)
+		}
+		podName = pods.Items[0].Name
+	}
+
+	return pod, nil
```
```suggestion
	pods, err := cs.CoreV1().Pods(namespace).List(
		context.Background(),
		metav1.ListOptions{LabelSelector: podSelector},
	)
	if err != nil {
		return "", fmt.Errorf("error listing pods: %w", err)
	}
	if len(pods.Items) == 0 {
		return "", fmt.Errorf("no pods found matching selector %q", podSelector)
	} else if len(pods.Items) > 1 {
		return "", fmt.Errorf("multiple pods found matching selector %q, use --pod instead", podSelector)
	}
	return pods.Items[0].Name, nil
```

jimmidyson commented on cmd/copy-file-to-pod/main.go:
```diff
@@ -109,6 +117,34 @@ func main() {
+		return "", err
+	}
+
+	var pod string
```
```suggestion
```

jimmidyson commented on cmd/copy-file-to-pod/main.go:
```diff
@@ -26,7 +26,7 @@ the destination file.
 Usage:
 
-    copy-file-to-pod [--kubeconfig file] [--namespace namespace] [--container container] file pod:/path
+    copy-file-to-pod [--kubeconfig file] [--namespace namespace] [--container container] [--pod-selector="app=remote-dst"] --pod pod file remote-path
```
```suggestion
    copy-file-to-pod [--kubeconfig file] [--namespace namespace] [--container container] (--pod-selector="app=remote-dst" | --pod=podName) file remote-path
```

dkoshkin commented:
Thanks for the suggestions, fixed those.

I also updated the demo to use this new functionality. Now its actually working:
```
node/self-hosted-registry-demo-control-plane condition met
node/self-hosted-registry-demo-worker condition met
NAME                                      STATUS   ROLES           AGE   VERSION
self-hosted-registry-demo-control-plane   Ready    control-plane   51s   v1.32.2
self-hosted-registry-demo-worker          Ready    <none>          37s   v1.32.2
pod "busybox" deleted
error: timed out waiting for the condition
Yay everything worked as expected - this was supposed to fail!
```
